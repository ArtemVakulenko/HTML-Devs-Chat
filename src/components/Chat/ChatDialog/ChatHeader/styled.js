import styled from 'styled-components';

export const StChatDialogHeader = styled.div`
    padding: 5px 10px;
    width: 100%;
    display: flex;
    flex-direction: row;
    align-items: center;
    background-color: ${({ colors, theme }) => colors[theme].secondary};
`;

export const StChatTitle = styled.div`
    margin-left: 25px;
    font-size: 2rem;
    display: flex;
    align-items: center;
    width: 90%;
`;  
