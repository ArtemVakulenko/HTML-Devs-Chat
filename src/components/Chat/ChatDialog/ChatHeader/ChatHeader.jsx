import React from 'react';
import PropTypes from 'prop-types';
import { useHistory } from 'react-router';
import Button from 'src/components/_common/Button';
import { useTheme } from 'src/components/hooks';
import { icons } from 'constants';
import { StChatDialogHeader, StChatTitle } from './styled';

const ChatHeader = ({ activeRoom, clearActiveRoom, clearMessageValue }) => {
  const history = useHistory();
  const { colors, theme } = useTheme();
  const handleClick = () => {
    history.push('/main');
    clearActiveRoom();
    clearMessageValue();
  };

  return (
    <StChatDialogHeader colors={colors} theme={theme}>
    <StChatTitle>{activeRoom.name}</StChatTitle>
    <Button icon={icons.closeModalDark} onClick={handleClick}/>
    </StChatDialogHeader>
  );
};

ChatHeader.propTypes = {
  activeRoom: PropTypes.object,
  clearActiveRoom: PropTypes.func,
  clearMessageValue: PropTypes.func,
};

export default ChatHeader;
