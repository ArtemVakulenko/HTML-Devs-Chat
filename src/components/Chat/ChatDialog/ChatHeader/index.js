import { connect } from 'react-redux';
import { activeRoom } from 'store/mainPage/selectors';
import { clearActiveRoom, clearMessageValue } from 'store/mainPage/actions';
import ChatHeader from './ChatHeader';

const mapStateToProps = (state) => ({
  activeRoom: activeRoom(state),
});
const mapDispatchToProps = {
  clearActiveRoom, clearMessageValue,
};

export default connect(mapStateToProps, mapDispatchToProps)(ChatHeader);
