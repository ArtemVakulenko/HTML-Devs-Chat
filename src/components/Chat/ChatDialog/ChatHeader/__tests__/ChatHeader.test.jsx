import React from 'react';
import configureStore from 'redux-mock-store';
import ChatHeader from '../ChatHeader';
import { shallowSmart } from '../../../../../__tests__/testHelper';

jest.mock('react-i18next', () => ({
  useTranslation: jest.fn().mockImplementation(() => ({
    t: jest.fn(),
  })),
}));
jest.mock('../../../../hooks', () => ({
  useTheme: jest.fn().mockImplementation(() => ({
    colors: {
      dark: {
        color: '',
        input: '',
        placeholder: '',
      },
    },
    theme: 'dark',
  })),
}));
const mockStore = configureStore();
const store = mockStore({
  sidebar: {
    search: '',
    rooms: [],
  },
  chat: {
    messages: [],
    roomId: '',
    messageValue: '',
    activeRoom: {},
  },
  user: {
    id: '',
    login: '',
    name: '',
    lastName: '',
    img: '',
    hobbie: '',
    city: '',
    age: '',
  },
});

jest.mock('react-notifications', () => ({
  NotificationManager: {
    error: jest.fn(),
    success: jest.fn(),
  },
}));
describe('ChatHEader', () => {
  describe('123', () => {
    const props = {
      clearMessageValue: jest.fn(),
      clearActiveRoom: jest.fn(),
      activeRoom: {},
    };
    it('Should match snapshot', () => {
      const component = shallowSmart(<ChatHeader {...props} />, store);
      expect(component.html()).toMatchSnapshot();
    });
  });
});
