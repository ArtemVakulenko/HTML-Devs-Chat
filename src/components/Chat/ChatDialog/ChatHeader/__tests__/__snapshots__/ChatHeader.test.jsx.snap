// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatHEader 123 Should match snapshot 1`] = `"<div class=\\"styled__StChatDialogHeader-frWLNJ iLTYqh\\"><div class=\\"styled__StChatTitle-erdfFm fRrVrb\\"></div><button class=\\"styled__StIcon-chPeWD giwZPn\\"></button></div>"`;

exports[`LoginPage 123 Should match snapshot 1`] = `"<div class=\\"styled__StChatDialogHeader-frWLNJ iLTYqh\\"><div class=\\"styled__StChatTitle-erdfFm fRrVrb\\"></div><button class=\\"styled__StIcon-chPeWD giwZPn\\"></button></div>"`;
